### PR TITLE
Telecommunications Updated Machine UI

### PR DESCRIPTION
**Pull request by WondaMegapon**
## About The Pull Request
Updates the interface for the telecommunication machines, allowing them to run using TGUI.

UpdatedTCommUI

## Why It's Good For The Game
It visually enhances the experience of telecomms and turns a once daunting interface into something players can easily approach.

## Changelog
:cl:
add: Telecommunication machines now use a TGUI based interface
del: Removed HTML based telecommunication machine code
balance: Telecommunication machines can have their interfaces viewed without a multitool
server: Telecommunication machines now log changed settings
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/Tcomm.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+                    disabled={!multibuff}
+                    onClick={() => act('flush')} />
+                </LabeledControls>
+              </Section>)}
+          </Box>  
+        )}
+      </Window.Content>
+    </Window>
+  );
+};
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Tcomm.js:245]: Missing final newline.

File: tgui/packages/tgui/interfaces/Tcomm.js [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+                      <NumberInput
+                        animate
+                        unit="GHz"
+                        step={0.2}
+                        stepPixelSize={10}
+                        minValue={minfreq / 10}
+                        maxValue={maxfreq / 10}
+                        value={frequency}
+                        onChange={(e, value) => act(
+                          "tempfreq", { value: value })}
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Tcomm.js:204]: You can use a shorthand syntax for properties with the same name as variable:
```{ value }```

File: tgui/packages/tgui/interfaces/Tcomm.js [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+                  <Table.Row key={frequencies.i} className="candystripe">
+                    <Table.Cell bold>
+                      {entry/10} GHz
+                    </Table.Cell>
+                    <Table.Cell>
+                      {RADIO_CHANNELS
+                        .find(channel => channel.freq === entry) && (
+                        <Box inline color={RADIO_CHANNELS
+                          .find(channel => channel.freq === entry)
+                          .color} ml={2}>
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Tcomm.js:161]: Each property should preferably be on its own line, to make it easier to read.

**Review by stylemistake — COMMENTED** [review]
Looks pretty damn sweet for a first PR.

**Comment by LemonInTheDark** [comment]
give this man good first pr for the love of god

**Comment by stylemistake** [comment]
@LemonInTheDark "Good First PR" is traditionally given out to first contributors with "barely passable" contributions :D.

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 4]
```diff
@@ -27,74 +29,176 @@
...
-					dat += "[format_frequency(x)] GHz<a href='?src=[REF(src)];delete=[x]'>\[X\]</a>; "
-				else
-					dat += "[format_frequency(x)] GHz<a href='?src=[REF(src)];delete=[x]'>\[X\]</a>"
-		else
-			dat += "NONE"
+/obj/machinery/telecomms/ui_interact(mob/user, datum/tgui/ui)
+	operator = user
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Tcomm")
```
Comment (stylemistake) [inline code/game/machinery/telecomms/machine_interactions.dm:36]: Change interface name to something proper, like `Telecomms`.

File: code/game/machinery/telecomms/machine_interactions.dm [hunk 5]
```diff
@@ -27,74 +29,176 @@
...
+		if("receive")
+			receiving = !receiving
+			. = TRUE
+
+/obj/machinery/telecomms/bus/add_act(action, params)
+	switch(action)
+		if("change_freq")
+			var/newfreq = params["value"]
+			if(newfreq)
+				if(findtext(num2text(newfreq), "."))
```
Comment (stylemistake) [inline code/game/machinery/telecomms/machine_interactions.dm:191]: What happens if you pass an integer frequency with no dot after a division by 10? I'd say just stick to one format of frequency, and where necessary, divide or multiply by 10.

**Comment by WondaMegapon** [comment]
Oops. Guess I'll have to figure out what went wrong in the morning.

**Review by stylemistake — DISMISSED** [review]
Looks good to me. 👍

**Pull request merged**